ambry-clustermap/src/main/java/com.github.ambry.clustermap/CloudServiceClusterChangeHandler.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.clustermap.HelixClusterManager.ClusterChangeHandlerCallback;
+import com.github.ambry.config.ClusterMapConfig;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Currently, this just adds a single {@link ReplicaType#CLOUD_BACKED} replica every time a partition is added to allow
+ * for requests to be routed to cloud services. Currently it is only registered as a {@link ClusterMapChangeListener},
+ * but once the clustermap natively supports VCR clusters, this class can implement {@link HelixAwareClusterChangeHandler} and
```

Comment:
Is there a doc briefly describing how to support VCR clusters in clustermap? I am interested in this part and probably can work on some of its implementation. Can I envision that clustermap will incorporate both CloudReplica (for server's two-way replication) and CloudServiceReplica (for router to issue request)?

Reply:
No, there is no up to date doc, but I think our end goal should be that ClusterMap provides full support for spectating vcr and server state such that we can unify `CloudToStoreReplicationManager` and `ReplicationManager` (since an ambry-server will soon have to be able to replicate from both servers and vcrs simultaneously).